src/ast/passes/semantic_analyser.cpp

```diff
@@ -2336,12 +2336,30 @@ void SemanticAnalyser::visit(Cast &cast)
 
   if (!cast.type.IsIntTy() && !cast.type.IsPtrTy() &&
       !(cast.type.IsPtrTy() && !cast.type.GetElementTy()->IsIntTy() &&
-        !cast.type.GetElementTy()->IsRecordTy()))
+        !cast.type.GetElementTy()->IsRecordTy()) &&
+      // we support casting integers to int arrays
+      !(cast.type.IsArrayTy() && cast.type.GetElementTy()->IsIntTy()))
   {
     LOG(ERROR, cast.loc, err_) << "Cannot cast to \"" << cast.type << "\"";
   }
-  if (cast.type.IsIntTy() && !rhs.IsIntTy() && !rhs.IsPtrTy() &&
-      !rhs.IsCtxAccess())
+
+  if (cast.type.IsArrayTy())
+  {
+    if (cast.type.GetElementTy()->IsBoolTy())
+    {
+      LOG(ERROR, cast.loc, err_) << "Bit arrays are not supported";
+      return;
+    }
+
+    if (rhs.IsIntTy())
+      cast.type.is_internal = true;
+  }
+
+  if ((cast.type.IsIntTy() && !rhs.IsIntTy() && !rhs.IsPtrTy() &&
```

Comment:
Is it useful to enable casting a pointer to an array? Would it make more sense to let the user do `(uint8[8])(uint64)$ptr`?

Reply:
Ah I misread. This is the pre-existing ptr -> int check

Reply:
Yeah, only int <-> array casts for now. I'm going to refactor this whole thing in a follow-up PR b/c it's becoming too hard to read (and it's even missing few checks).